src/components/RelicModal.tsx

```diff
@@ -440,3 +463,40 @@ function SubstatInput(props: { index: number; upgrades: RelicUpgradeValues[]; re
     </Flex>
   )
 }
+
+function reformatRelic(relic) {
+  const thing = {
+    grade: relic.grade,
+    enhance: relic.enhance,
+    part: relic.part,
+    set: relic.set,
+    mainStatType: renderMainStat(relic).stat,
+    substatType0: renderSubstat(relic, 0).stat,
+    substatValue0: renderSubstat(relic, 0).value,
+    substatType1: renderSubstat(relic, 1).stat,
+    substatValue1: renderSubstat(relic, 1).value,
+    substatType2: renderSubstat(relic, 2).stat,
+    substatValue2: renderSubstat(relic, 2).value,
+    substatType3: renderSubstat(relic, 3).stat,
+    substatValue3: renderSubstat(relic, 3).value,
+  }
+  return thing
+}
+
+function compareRelics(relic, final) {
+  const initial = reformatRelic(relic)
+  if (initial.grade != final.grade) return true
+  if (initial.enhance != final.enhance) return true
+  if (initial.part != final.part) return true
+  if (initial.set != final.set) return true
+  if (initial.mainStatType != final.mainStatType) return true
+  if (initial.substatType0 != final.substatType0) return true
+  if (initial.substatValue0 != final.substatValue0) return true
+  if (initial.substatType1 != final.substatType1) return true
+  if (initial.substatValue1 != final.substatValue1) return true
+  if (initial.substatType2 != final.substatType2) return true
+  if (initial.substatValue2 != final.substatValue2) return true
+  if (initial.substatType3 != final.substatType3) return true
+  if (initial.substatValue3 != final.substatValue3) return true
+  return false
+}
```

Comment:
We have a Util that does this sort of calculation `Utils.objectHash`, which can just convert these into hash comparisons. 

Also probably bad practice to compare stats with a render function, if we ever change how stats are rendered then that may break relic comparisons.